File: python/claude/p12_nr_ver2.py
import sys

def getValueFromArray(arr, size, index):
    if not arr or size <= 0 or index < 0 or index >= size:
        return -1
    return arr[index]

def main():
    arr = [10, 20, 30, 40, 50]
    size = len(arr)

    try:
        line = input().strip()
        index = int(line)
    except (ValueError, EOFError):
        sys.exit(255)

    result = getValueFromArray(arr, size, index)
    sys.exit(255 if result == -1 else result)

if __name__ == "__main__":
    main()

File: javascript/claude/p12_nr_ver2.js
const readline = require("readline");

function getValueFromArray(arr, size, index) {
    if (!arr || size <= 0 || index < 0 || index >= size)
        return -1;
    return arr[index];
}

const rl = readline.createInterface({ input: process.stdin });

rl.on("line", (line) => {
    rl.close();

    const arr = [10, 20, 30, 40, 50];
    const size = arr.length;

    const parsed = parseInt(line.trim(), 10);
    if (isNaN(parsed)) {
        process.exit(255);
    }

    const result = getValueFromArray(arr, size, parsed);
    process.exit(result === -1 ? 255 : result);
});